migrations/message_store/00008_updatePrimaryKey_add_col.up.sql

```diff
@@ -0,0 +1,32 @@
+ALTER TABLE message RENAME TO message_backup;
+
+CREATE TABLE IF NOT EXISTS message (
+  pubsubTopic BLOB NOT NULL,
+  contentTopic BLOB NOT NULL,
+  payload BLOB,
+  version INTEGER NOT NULL,
+  timestamp INTEGER NOT NULL,
+  id BLOB,
+  messageHash BLOB, -- Newly added, this will be populated with a counter value
+  storedAt INTEGER NOT NULL,
+  CONSTRAINT messageIndex PRIMARY KEY (storedAt, messageHash)
+) WITHOUT ROWID;
+
+
+INSERT INTO message(pubsubTopic, contentTopic, payload, version, timestamp, id, messageHash, storedAt)
+SELECT
+  mb.pubsubTopic,
+  mb.contentTopic,
+  mb.payload,
+  mb.version,
+  mb.timestamp,
+  mb.id,
+  (
+    SELECT COUNT(*)
+    FROM message_backup AS mb2
+    WHERE mb2.storedAt <= mb.storedAt
+  ) as messageHash, -- to populate the counter values
```

Comment:
so basically here in the case of migration, we can not leave the newly added `messageHash` as Null or empty for backdated messages/DB-rows, so using this script we are filling `messageHash` using a counter variable. For eg. If there were 50 messages before running the migration scripts, then after migration `messageHash` will be introduced so for the previous 50 messages in the DB we fill the `messageHash` value for those as a counter from 1 to 50. This way unique value will be there in `messageHash` column. WDYT @Ivansete-status

Reply:
I wonder if this would take long in a DB with large amount of rows like those from status.

Perhaps this https://www.sqlite.org/lang_corefunc.html#randomblob could be a good alternative to doing a count for each row inserted, or perhaps using https://www.sqlite.org/c3ref/create_function.html and defining a function in nim to calculate the message hash that can be called from sqlite.  (this is for sure much more complicated to implement)

Reply:
> I wonder if this would take long in a DB with large number of rows like those from status.
> 
> Perhaps this https://www.sqlite.org/lang_corefunc.html#randomblob could be a good alternative to doing a count for each row inserted, or perhaps using https://www.sqlite.org/c3ref/create_function.html and defining a function in nim to calculate the message hash that can be called from sqlite. (this is for sure much more complicated to implement)

I think the current solution is capable of scaling, and it is a simpler approach to the issue here, since post-migration, there will be long strings of hashes in `messageHash` column, so basically backfilling DB with counter values will be unique and less resource-consuming in comparison?
WDYT @jm-clius @richard-ramos @Ivansete-status

Reply:
> > I wonder if this would take long in a DB with large number of rows like those from status.
> > Perhaps this https://www.sqlite.org/lang_corefunc.html#randomblob could be a good alternative to doing a count for each row inserted, or perhaps using https://www.sqlite.org/c3ref/create_function.html and defining a function in nim to calculate the message hash that can be called from sqlite. (this is for sure much more complicated to implement)
> 
> I think the current solution is capable of scaling, and it is a simpler approach to the issue here, since post-migration, there will be long strings of hashes in `messageHash` column, so basically backfilling DB with counter values will be unique and less resource-consuming in comparison? WDYT @jm-clius @richard-ramos @Ivansete-status

Actually, @richard-ramos  is right, the `count(*)` operation will be more intensive than `randomBlob(N)`, so it will be nice to make this change. But make sure that this function is available in clients older this SQLite version [3.3.13](https://www.sqlite.org/releaselog/3_3_13.html). 

Also as recommended by @jm-clius, a counter-mechanism i.e. using simple values 1,2,3... will not be the case then.

Reply:
Note that this migration script should be tailored to what Status's fleets need for their 30-day SQLite stores. In other words, all other Store nodes that I'm aware of have much shorter retention periods. I'd go for whichever function is fastest (`randomBlob(N)`?). It should be easy to determine which version of SQLite is running on the Status Store nodes.

Reply:
considering that, SQLite 3.3.13 was launched in 2007, we should proceed with `randomBlob(N)` without version check?

Reply:
No, it should be possible to check the SQLite version in the ABI used by the version of nwaku currently deployed to the fleets.

Reply:
Seems like `3.40`, but probably worth checking history here for any recent upgrades. https://raw.githubusercontent.com/arnetheduck/nim-sqlite3-abi/362e1bd9f689ad9f5380d9d27f0705b3d4dfc7d3/sqlite3_abi/sqlite3.c

Reply:
Would it be possible to just set this attribute to "empty" or nil ?
It doesn't make sense from my point of view to add this complexity because, in the end, we are not calculating the messageHash in the same way as the code base is doing. Or am I missing something?

Reply:
> No, it should be possible to check the SQLite version in the ABI used by the version of nwaku currently deployed to the fleets.

One question: the `randomBlob()` is there to begin with from tag v0.1 so it is safer to assume that we can use randomBlob that even on the earliest clients it is available and will not break any backward/outdated clients?